packages/template-retail-react-app/app/hooks/use-dnt-notification.js

```diff
@@ -0,0 +1,176 @@
+    useDisclosure,
+    Heading,
+    Stack,
+    Text,
+    Flex
+} from '@salesforce/retail-react-app/app/components/shared/ui'
+import {HideOnDesktop, HideOnMobile} from '@salesforce/retail-react-app/app/components/responsive'
+import {useDNT} from '@salesforce/commerce-sdk-react'
+import {useLocation} from 'react-router-dom'
+
+export const DntNotification = ({isOpen, onOpen, onClose}) => {
+    const {dntStatus, updateDNT} = useDNT()
+    const {formatMessage} = useIntl()
+    const location = useLocation()
+
+    useEffect(() => {
+        if (dntStatus === undefined) {
+            onOpen()
+        } else {
+            onClose()
+        }
+    }, [location, dntStatus])
+
+    const onCloseNotification = async () => {
+        await updateDNT(null)
+        onClose()
+    }
+
+    const buttons = (
+        <>
+            <Button
+                bg="white"
+                color="black"
+                border="1px"
+                _hover={{bg: 'gray.100'}}
+                borderColor="gray.100"
+                boxShadow="md"
+                onClick={async () => {
+                    await updateDNT(true)
+                    onClose()
+                }}
+                aria-label={formatMessage({
+                    id: 'dnt_notification.button.assistive_msg.decline',
+                    defaultMessage: 'Decline Tracking'
+                })}
+                width="100%"
+            >
+                <FormattedMessage defaultMessage="Decline" id="dnt_notification.button.decline" />
+            </Button>
+            <Button
+                onClick={async () => {
+                    await updateDNT(false)
+                    onClose()
+                }}
+                boxShadow="md"
+                width="100%"
+                aria-label={formatMessage({
+                    id: 'dnt_notification.button.assistive_msg.accept',
+                    defaultMessage: 'Accept Tracking'
+                })}
+            >
+                <FormattedMessage defaultMessage="Accept" id="dnt_notification.button.accept" />
+            </Button>
+        </>
+    )
+    // Placeholder for the consent tracking form for demonstration purposes
+    const description = (
+        <Text color={'gray.700'} fontWeight={500} marginTop={7}>
+            <FormattedMessage
+                defaultMessage="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur."
+                id="dnt_notification.description"
+            />
+        </Text>
+    )
+
+    return (
+        <Modal
+            size="sm"
+            data-testid="sf-dnt-notification"
+            blockScrollOnMount={false}
+            closeOnOverlayClick={false}
+            trapFocus={false}
```

Comment:
It's cool that we can customize the Modal to behave more like a notification and the page content is still interactive 👍